library/src/main/java/org/apache/apex/malhar/lib/fs/s3/S3Reconciler.java

```diff
@@ -161,37 +169,46 @@ protected void processCommittedData(FSRecordCompactionOperator.OutputMetaData ou
-      logger.debug("found metaData = {}", metaData);
-      committedTuples.remove(metaData);
-      try {
-        Path dest = new Path(metaData.getPath());
-        //Deleting the intermediate files and when writing to tmp files
-        // there can be vagrant tmp files which we have to clean
-        FileStatus[] statuses = fs.listStatus(dest.getParent());
-
-        for (FileStatus status : statuses) {
-          String statusName = status.getPath().getName();
-          if (statusName.endsWith(TMP_EXTENSION) && statusName.startsWith(metaData.getFileName())) {
-            //a tmp file has tmp extension always preceded by timestamp
-            String actualFileName = statusName.substring(0,
-                statusName.lastIndexOf('.', statusName.lastIndexOf('.') - 1));
-            logger.debug("actualFileName = {}", actualFileName);
-            if (metaData.getFileName().equals(actualFileName)) {
-              logger.debug("deleting stray file {}", statusName);
-              fs.delete(status.getPath(), true);
-            }
-          } else if (statusName.equals(metaData.getFileName())) {
-            logger.info("deleting s3-compaction file {}", statusName);
+      removeIntermediateFiles(doneTuples.poll());
+    }
+  }
+
+  /**
+   * Remove intermediate files
+   */
+  protected void removeIntermediateFiles(FSRecordCompactionOperator.OutputMetaData metaData)
+  {
+    logger.debug("found metaData = {}", metaData);
+    committedTuples.remove(metaData);
+    try {
+      Path dest = new Path(metaData.getPath());
+      //Deleting the intermediate files and when writing to tmp files
+      // there can be vagrant tmp files which we have to clean
+      FileStatus[] statuses = fs.listStatus(dest.getParent());
+
+      for (FileStatus status : statuses) {
+        String statusName = status.getPath().getName();
+        if (statusName.endsWith(TMP_EXTENSION) && statusName.startsWith(metaData.getFileName())) {
+          //a tmp file has tmp extension always preceded by timestamp
+          String actualFileName = statusName.substring(0,
+              statusName.lastIndexOf('.', statusName.lastIndexOf('.') - 1));
+          logger.debug("actualFileName = {}", actualFileName);
+          if (metaData.getFileName().equals(actualFileName)) {
+            logger.debug("deleting stray file {}", statusName);
             fs.delete(status.getPath(), true);
           }
+        } else if (statusName.equals(metaData.getFileName())) {
+          logger.info("deleting s3-compaction file {}", statusName);
+          fs.delete(status.getPath(), true);
         }
-      } catch (IOException e) {
-        logger.error("Unable to Delete a file: {}", metaData.getFileName());
       }
+    } catch (IOException e) {
+      logger.error("Unable to Delete a file: {}", metaData.getFileName());
     }
+    // Emit the meta data with S3 path
+    metaData.setPath(getDirectoryName() + Path.SEPARATOR + metaData.getFileName());
+    outputPort.emit(metaData);
```

Comment:
Emit only if port is connected; else, tuples will be in bufferserver for no one to consume.